src/main/java/com/mraof/minestuck/client/gui/DialogueButton.java

```diff
@@ -0,0 +1,105 @@
+package com.mraof.minestuck.client.gui;
+
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.sounds.SoundEvents;
+import net.minecraft.util.FormattedCharSequence;
+import net.minecraftforge.client.gui.widget.ExtendedButton;
+
+import java.util.List;
+
+@MethodsReturnNonnullByDefault
+public class DialogueButton extends ExtendedButton
```

Comment:
Looks like you're not making much use of extending specifically `ExtendedButton`, since you're fully overriding `renderWidget()`.
You're better off extending `Button` instead. *Maybe* even as far as extending `AbstractButton` instead of `Button`?